x-pack/plugins/enterprise_search/server/collectors/app_search/telemetry.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { set } from 'lodash';
+import { ISavedObjectsRepository, SavedObjectsServiceStart } from 'src/core/server';
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+
+import { AS_TELEMETRY_NAME, ITelemetrySavedObject } from '../../saved_objects/app_search/telemetry';
+
+/**
+ * Register the telemetry collector
+ */
+
+interface Dependencies {
+  savedObjects: SavedObjectsServiceStart;
+  usageCollection: UsageCollectionSetup;
+}
+
+export const registerTelemetryUsageCollector = ({
+  usageCollection,
+  savedObjects,
+}: Dependencies) => {
+  const upgradeAssistantUsageCollector = usageCollection.makeUsageCollector({
+    type: AS_TELEMETRY_NAME,
+    fetch: async () => fetchTelemetryMetrics(savedObjects),
+  });
+  usageCollection.registerCollector(upgradeAssistantUsageCollector);
+};
+
+/**
+ * Fetch the aggregated telemetry metrics from our saved objects
+ */
+
+const fetchTelemetryMetrics = async (savedObjects: SavedObjectsServiceStart) => {
+  const savedObjectsRepository = savedObjects.createInternalRepository();
```

Comment:
https://github.com/elastic/kibana/blob/a2c002d13030d5db2252d00983927ab8e21fe7e9/docs/development/core/server/kibana-plugin-core-server.savedobjectsservicestart.md#properties

createInternalRepository vs createScopedRepository

Since we're using internal and not scoped, these will not be saved at a user level, but instead be tracked across all usage.

As I'm writing that I'm realizing that is probably exactly what we want so this point is moot. Will let this in case someone finds it interesting to see the distinction pointed out in the docs.

Reply:
Looks like these are checked in `plugin.ts` 👌

Reply:
Oh snap, I hadn't dug into the docs (which I'll admit I find very hard to navigate) enough to realize there were different repository types. Thanks for finding and clearing that distinction!